onnxruntime/core/framework/graph_partitioner.cc

```diff
@@ -865,6 +866,36 @@ static Status CreateEpContextModel(const ExecutionProviders& execution_providers
   auto& ep_graph = ep_context_model.MainGraph();
   ep_graph.SetDescription(graph.Description());
 
+  // Generate EP compatibility strings for OrtEp types and add to model metadata
+  {
+    const GraphViewer graph_viewer(graph);
+    for (const auto& ep : execution_providers) {
+      try {
+        // Generate the compatibility string for this EP
+        std::string compatibility_string = ep->GetCompiledModelCompatibilityInfo(graph_viewer);
```

Comment:
I'm not how easily an EP can utilize the model that it receives in this call to GetCompiledModelCompatibilityInfo(). This model is in a state between the original uncompiled model and the finale EPContext model. Specifically, the nodes assigned to a compiling EP are currently fused into a generic "fused node" with a computation function (i.e., it is not an EPContext node yet).

Would it make more sense to have this logic at the end of `CreateEpContextModel`, once all generic "fused nodes" have been replaced with proper EPContext nodes?